File: AXIOMHIVE/src/agents/cerebrum.py
# @AXIOMHIVE @DEVDOLLZ ALEXIS ADAMS
# Agent: Cerebrum (Strategic Diagnosis Core)

from typing import Dict

class Cerebrum:
    """Deconstructs intent and maps it to strategic objectives."""

    def __init__(self):
        # AXIOM: NOISE=-∞
        self._noise_words = {"please", "can", "you", "design", "a", "an", "the", "for", "of"}

    def filter_noise(self, directive: str) -> str:
        """Purges non-essential tokens to reveal core intent."""
        tokens = directive.lower().split()
        filtered_tokens = [t for t in tokens if t not in self._noise_words]
        return " ".join(filtered_tokens)

    def diagnose_strategy(self, clean_directive: str) -> Dict[str, str]:
        """Analyzes intent to identify the strategic imperative and complexity type."""
        # This is a simplified model; a production version would use NLP.
        strategy = {"primary_moat": "None", "complexity_type": "None", "core_task": "Unknown"}
        
        if "data moat" in clean_directive or "feedback loop" in clean_directive:
            strategy["primary_moat"] = "Data"
            strategy["core_task"] = "GENERATE_DATA_LOOP_ARCHITECTURE"
        elif "simplify" in clean_directive or "reduce complexity" in clean_directive:
            strategy["primary_moat"] = "Operational"
            strategy["core_task"] = "GENERATE_SIMPLIFICATION_WORKFLOW"
            
        if "system" in clean_directive or "architecture" in clean_directive:
            strategy["complexity_type"] = "System/Technical"
        elif "process" in clean_directive:
            strategy["complexity_type"] = "Process"
            
        print(f"CEREBRUM :: Intent diagnosed. Moat Target={strategy['primary_moat']}.")
        return strategy

File: AXIOMHIVE/src/agents/cerebrum.rs
// @AXIOMHIVE @DEVDOLLZAI
// Cerebrum: Deconstructs intent and maps it to strategic objectives.

use serde::{Deserialize, Serialize};
use anyhow::Result;

#[derive(Serialize, Deserialize, Debug)]
pub struct Strategy {
    pub primary_moat: String,
    pub complexity_type: String,
    pub core_task: String,
}

pub struct Cerebrum {
    noise_words: Vec<String>,
}

impl Cerebrum {
    pub fn new() -> Self {
        Cerebrum {
            noise_words: vec![
                "please".to_string(),
                "can you".to_string(),
                "design".to_string(),
                "a".to_string(),
                "an".to_string(),
                "the".to_string(),
                "for".to_string(),
                "of".to_string(),
                "to".to_string(),
            ],
        }
    }

    pub fn filter_noise(&self, directive: &str) -> String {
        let tokens: Vec<String> = directive
            .to_lowercase()
            .split_whitespace()
            .map(|s| s.to_string())
            .collect();
        let filtered: Vec<String> = tokens
            .into_iter()
            .filter(|t| !self.noise_words.contains(t))
            .collect();
        filtered.join(" ")
    }

    pub fn diagnose_strategy(&self, clean_directive: &str) -> Result<Strategy> {
        let mut strategy = Strategy {
            primary_moat: "None".to_string(),
            complexity_type: "None".to_string(),
            core_task: "Unknown".to_string(),
        };

        if clean_directive.contains("data moat") || clean_directive.contains("feedback loop") {
            strategy.primary_moat = "Data".to_string();
            strategy.core_task = "GENERATE_DATA_LOOP_ARCHITECTURE".to_string();
        } else if clean_directive.contains("simplify") || clean_directive.contains("reduce complexity") {
            strategy.primary_moat = "Operational".to_string();
            strategy.core_task = "GENERATE_SIMPLIFICATION_WORKFLOW".to_string();
        }

        if clean_directive.contains("architecture") {
            strategy.complexity_type = "System/Technical".to_string();
        } else if clean_directive.contains("process") {
            strategy.complexity_type = "Process".to_string();
        }

        println!("CEREBRUM :: Intent diagnosed. Moat Target={}.", strategy.primary_moat);
        Ok(strategy)
    }
}